sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -340,111 +356,80 @@ void createUri(BlockStmt body, HttpRequestContext method) {
-                        body.addStatement("queryParamMap.put(" + keyExpr + ", " + valueCode + ");");
+                        valueExpression = new StringLiteralExpr(value);
                     }
                 } else {
-                    // Static/literal value: do NOT escape key or value
-                    // If the value is an empty string, treat as "", if null, treat as null
-                    if (valueExpr == null) {
-                        body.addStatement("queryParamMap.put(\"" + key + "\", null);");
-                    } else {
-                        valueCode = "\"" + valueExpr.replace("\"", "\\\"") + "\"";
-                        body.addStatement("queryParamMap.put(\"" + key + "\", " + valueCode + ");");
-                    }
+                    // For non-static query parameters the value should be the name of the method parameter.
+                    valueExpression = StaticJavaParser.parseExpression(value);
                 }
             } else {
-                // Multiple values: always add as List<String>
-                String joinedValues = CodeGenUtils.toJavaArrayInitializer(values, true);
-                boolean isParam = values.stream()
-                    .allMatch(v -> method.getParameters().stream().anyMatch(p -> p.getName().equals(v)));
-                String keyExpr = isParam ? "UriEscapers.QUERY_ESCAPER.escape(\"" + key + "\")" : "\"" + key + "\"";
-
-                if (!joinedValues.trim().isEmpty()) {
-                    body.tryAddImportToParentCompilationUnit(Arrays.class);
-                    if (shouldEncode && isParam) {
-                        body.tryAddImportToParentCompilationUnit(UriEscapers.class);
-                        body.addStatement("queryParamMap.put(" + keyExpr + ", Arrays.asList(" + joinedValues + ")"
-                            + ".stream().map(UriEscapers.QUERY_ESCAPER::escape).collect(Collectors.toList()));");
-                    } else {
-                        // For static query params, do NOT escape key or value
-                        body.addStatement("queryParamMap.put(" + keyExpr + ", Arrays.asList(" + joinedValues + "));");
-                    }
-                } else {
-                    // if joinedValues is empty, put null
-                    body.addStatement("queryParamMap.put(\"" + key + "\", null);");
-                }
+                body.tryAddImportToParentCompilationUnit(Arrays.class);
+                valueExpression = StaticJavaParser
+                    .parseExpression("Arrays.asList(" + CodeGenUtils.toJavaArrayInitializer(values, true) + ")");
             }
-        });
 
-        // Append query parameters to the URL
-        body.tryAddImportToParentCompilationUnit(CoreUtils.class);
+            body.tryAddImportToParentCompilationUnit(GeneratedCodeUtils.class);
+            MethodCallExpr addParameterCall
+                = new MethodCallExpr(new NameExpr("GeneratedCodeUtils"), "addQueryParameter")
```

Comment:
Could you add a generation that shows how this method call looks like, that would be helpful to see its use case.

Reply:
I'm a little confused on what a generation is, is this an ask to add documentation here for what the generated code will look like?